Spatial_searching/include/CGAL/Point_container.h

```diff
@@ -55,6 +56,97 @@ class Point_container {
   // box of points
 
 public:
+  using Construct_cartesian_const_iterator_d = typename Traits::Construct_cartesian_const_iterator_d;
+  using FTP = typename Construct_cartesian_const_iterator_d::result_type;
+  using References = std::vector< std::vector<FTP> >;
+  using Ref_ptr = std::shared_ptr<References>;
+
+private:
+  Ref_ptr m_ref_ptr;
+
+  long m_start = -1;
+  long m_end = -1;
+
+  std::size_t m_dim = static_cast<std::size_t>(-1);
+  std::size_t m_depth = static_cast<std::size_t>(-1);
+
+  bool m_is_last_call = false;
+
+public:
+  struct Key_compare {
+
+    FT operator()(
+      const FTP& p, const FTP& q, // refs to points p and q
+      const std::size_t axis, // axis offset in (x, y, z, w, ...)
+      const std::size_t dim) const { // dimension
+
+      CGAL_assertion(p != FTP());
+      CGAL_assertion(q != FTP());
+      FT coords_diff = FT(0);
+      for (std::size_t k = 0; k < dim; ++k) {
+        const std::size_t idx = (k + axis) % dim; // stay inside dim
+        coords_diff = p[idx] - q[idx];
+        if (coords_diff != FT(0)) break;
+      }
+      return coords_diff;
```

Comment:
Should the result not be of type `bool`  with `true` for the first parameter being larger than the second?

Reply:
Not really, because we need to know also when the super keys are equal that is `coords_diff = 0`.